test_data/leave_sync.json

```diff
@@ -188,7 +187,6 @@
                                 "membership": "join"
                             },
                             "event_id": "$152034824468gOeNB:localhost",
-                            "membership": "join",
```

Comment:
Not sure how this got in or if this is ok according to the spec but it broke ruma's ser/de

Reply:
I think this might be some old event format, while it's fine to ignore it I don't think Ruma should break if it finds such a field.

Reply:
I'm surprised `serde` fails to deserialize when it finds extra keys. Before ruma-events had optional room_id fileds this might make more sense?

Reply:
I agree, there isn't much security risk in ignoring fields. I wonder if serde has an option to not fail unless a field that should be there isn't?

Reply:
Ignoring them is required for forwards compatibility. Serde should already do that by default: https://serde.rs/container-attrs.html#deny_unknown_fields

Reply:
This is probably because we don't use `serde`s `Deserialize` derive and missed this case when generating `Deserialize` implementations in ruma-events-macros. I would suspect a `#[serde(other)]` variant for our [`Field` enum](https://github.com/ruma/ruma/blob/master/ruma-events-macros/src/event.rs#L215-L221) fixes this.